### PR TITLE
Dont destroy the platform message loop on the Mac because the message pump is still not collected when the embedder state goes away.

### PR DESCRIPTION
**Pull request by chinmaygarde**
Fixes https://github.com/flutter/flutter/issues/3151

### CODE REVIEW

**Comment by chinmaygarde** [comment]
oops, duplicate import.

**Comment by abarth** [comment]
LGTM

File: sky/shell/platform/mac/platform_mac.mm [hunk 1]
```diff
@@ -83,17 +88,23 @@ static void RedirectIOConnectionsToSyslog() {
     sky::shell::Shell::InitStandalone(icu_data_path);
   }
 
+  ~EmbedderState() {
+    CHECK([NSThread isMainThread])
+        << "Embedder destruction must occur on the main platform thread";
+#if !TARGET_OS_IPHONE
+    embedder_message_loop_->QuitNow();
+    embedder_message_loop_.release();
```
Comment (jason-simmons) [inline sky/shell/platform/mac/platform_mac.mm]: In this case, how is embedder_message_loop_ deleted?

Comment (chinmaygarde) [inline sky/shell/platform/mac/platform_mac.mm]: We leak the instance. Though there is a `MessagePumpMac::Create()`, there is no corresponding way to collect that instance. And there is an assertion in the destructor of the message loop that trips on that check.

**Pull request merged**